src/Symfony/Component/FeatureFlag/FeatureChecker.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\FeatureFlag;
+
+use Symfony\Component\FeatureFlag\Exception\FeatureNotFoundException;
+
+final class FeatureChecker implements FeatureCheckerInterface
+{
+    private array $cache = [];
+
+    public function __construct(
+        private readonly FeatureRegistryInterface $featureRegistry,
+        private readonly mixed $default = false,
+    ) {
+    }
+
+    public function isEnabled(string $featureName, mixed $expectedValue = true): bool
+    {
+        return $this->getValue($featureName) === $expectedValue;
```

Comment:
It's probably not enough. Let's say I want to check whether a feature applies to a request object, which contains information for comparison (IP, user ID, etc.). How can I achieve that with this default checker?

Reply:
Since a feature can be a invokable class, these information can be injected (with the RequestStack in a Symfony application).

An alternative could be to pass a context to the feature (related to https://github.com/symfony/symfony/pull/53213#issuecomment-1876076459).

Reply:
Yes, but I'm afraid the injection method doesn't work for all cases, such as CLI and async processing. In those contexts, there is no Request.

>An alternative could be to pass a context to the feature (related to https://github.com/symfony/symfony/pull/53213#issuecomment-1876076459).

I don't see any other way to do this.

Reply:
Would a ContextAware / RequestAwareFeatureChecker be a possibility ?

Reply:
> Yes, but I'm afraid the injection method doesn't work for all cases, such as CLI and async processing. In those contexts, there is no Request.

I don't see this as an issue. What you have through Request is not needed in CLI such as client IP, headers, or such. Other possibilities can be injected AFAIK. 

Regardin a context we agree taht at some point it may be needed. But to keep things simple at start and not go with a badly design context I suggest to skip this feature ATM.

Reply:
> Yes, but I'm afraid the injection method doesn't work for all cases, such as CLI and async processing. In those contexts, there is no Request.

I don't see this as an issue. What you have through Request is not needed in CLI such as client IP, headers, or such. Other possibilities can be injected AFAIK. 

Regardin a context we agree taht at some point it may be needed. But to keep things simple at start and not go with a badly design context I suggest to skip this feature ATM.

Reply:
I agree about keeping it simple at first and avoid the `$context` completely if possible. For now one can always decorate the feature checker to perform a different check + pass any compound as `$expectedValue`.